src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,95 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd?: (movie: Movie) => void,
+};
+
+export const NewMovie = ({ onAdd = () => { } }: Props) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
   const [count] = useState(0);
+  const [isFilled, setIsFilled] = useState(false);
+
+  const emptyMovie = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+
+  const [newMovie, setNewMovie] = useState(emptyMovie);
+
+  const handleOnChange = (value: string, name: string) => {
+    setNewMovie({
+      ...newMovie,
+      [name]: value,
+    });
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onAdd(newMovie);
+    setNewMovie(emptyMovie);
+  };
+
+  useEffect(() => {
+    const isFormFilled
+      = newMovie.title.length > 0
+      && newMovie.imgUrl.length > 0
```

Comment:
it will work for an empty string either
```suggestion
      && newMovie.imgUrl
```